.pre-commit-config.yaml

```diff
@@ -8,11 +8,11 @@ repos:
           - --fix
           - --exit-non-zero-on-fix
           - --line-length=120
-          - --ignore=E731,E501,W605
+          - --ignore=E731,E501,W605,UP038
```

Comment:
Personal preference, or a special issue with the code base?

Reply:
https://docs.astral.sh/ruff/rules/non-pep604-isinstance/

Reply:
I was wondering why ignore it, not what the it is.

Reply:
I do think the link answers that 😉 

[Why is this bad?](https://docs.astral.sh/ruff/rules/non-pep604-isinstance/#why-is-this-bad)

Since Python 3.10, isinstance and issubclass can be passed a |-separated union of types, which is consistent with the union operator introduced in [PEP 604](https://peps.python.org/pep-0604/).

Note that this results in slower code. Ignore this rule if the performance of an isinstance or issubclass check is a concern, e.g., in a hot loop